webpack/assets/javascripts/react_app/components/common/AutocompleteInput/__tests__/AutocompleteInput.test.js

```diff
@@ -0,0 +1,201 @@
+      fireEvent.click(input);
+
+      await waitFor(() => {
+        const option = screen.getByText('Option 2');
+        fireEvent.click(option);
+      });
+
+      expect(input).toHaveValue('Option 2');
+    });
+
+    test('closes dropdown after selection', async () => {
+      render(<AutocompleteInput {...defaultProps} />);
+
+      const input = screen.getByRole('combobox');
+      fireEvent.click(input);
+
+      await waitFor(() => {
+        const option = screen.getByText('Option 1');
+        fireEvent.click(option);
+      });
+
+      await waitFor(() => {
+        expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Keyboard Navigation', () => {
+    test('opens dropdown on Enter key', async () => {
+      render(<AutocompleteInput {...defaultProps} />);
+
+      const input = screen.getByRole('combobox');
+      fireEvent.keyDown(input, { key: 'Enter' });
+
+      await waitFor(() => {
+        expect(screen.getByRole('listbox')).toBeInTheDocument();
+      });
+    });
+
+    test('navigates options with arrow keys', async () => {
+      render(<AutocompleteInput {...defaultProps} />);
+
+      const input = screen.getByRole('combobox');
+      fireEvent.click(input);
+
+      fireEvent.keyDown(input, { key: 'ArrowDown' });
+      await waitFor(() => {
+        expect(screen.getByText('Option 1')).toHaveClass(
+          'pf-v5-c-menu__item-text'
+        );
+      });
+    });
+  });
+
+  describe('Edge Cases', () => {
+    test('handles empty options array', async () => {
+      render(<AutocompleteInput {...defaultProps} options={[]} />);
+
+      const input = screen.getByRole('combobox');
+      fireEvent.click(input);
+
+      await waitFor(() => {
+        expect(screen.getByText('No matches found')).toBeInTheDocument();
+      });
+    });
+
+    test('handles undefined selected value', () => {
+      render(<AutocompleteInput {...defaultProps} selected={undefined} />);
+
+      const input = screen.getByRole('combobox');
+      expect(input).toHaveValue('');
+    });
+
+    test('handles numeric selected value', () => {
+      render(<AutocompleteInput {...defaultProps} selected={123} />);
+
+      const input = screen.getByRole('combobox');
+      expect(input).toHaveValue('123');
+    });
+
+    test('handles boolean selected value', () => {
+      render(<AutocompleteInput {...defaultProps} selected />);
+
+      const input = screen.getByRole('combobox');
+      expect(input).toHaveValue('true');
+    });
+  });
+
+  describe('Accessibility', () => {
+    test('has proper ARIA attributes when focused', async () => {
+      render(<AutocompleteInput {...defaultProps} />);
+
+      const input = screen.getByRole('combobox');
+      fireEvent.click(input);
+
+      await waitFor(() => {
+        fireEvent.keyDown(input, { key: 'ArrowDown' });
+        expect(input).toHaveAttribute('aria-activedescendant');
+      });
+    });
+  });
+});
```

Comment:
```suggestion
});

```
Nitpick: missing newline.